lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -10,6 +10,14 @@
 from pbench.server.database.models.datasets import Dataset
 
 
+def urlencode_query(query: Dict[str, Any]) -> str:
+    """We need to encode the metadata list as expected."""
+    new_query = {}
+    for k, v in sorted(query.items()):
+        new_query[k] = ",".join(v) if k == "metadata" else v
+    return urlencode(new_query)
+
+
```

Comment:
Can we import this directly from datasets_list.py?

Reply:
I will resolve this conversation in case we are moving forward with the merge and this does not delay it.

Reply:
I debated about this.  We could use the same method in the test, but then I'd have to write a separate test for the method which verifies the encoding.

Maybe that is better to do...

Reply:
Check out the new test.